erpnext/stock/landed_taxes_and_charges_common.js

```diff
@@ -30,7 +30,9 @@ document_list.forEach((doctype) => {
 
 			if (row.account_currency == company_currency) {
 				row.exchange_rate = 1;
+				frm.set_df_property('taxes', 'hidden', 1, row.name, 'exchange_rate');
 			} else if (!row.exchange_rate || row.exchange_rate == 1) {
+				frm.set_df_property('taxes', 'hidden', 0, row.name, 'exchange_rate');
```

Comment:
you can remove these two for now as they aren't really doing anything.
Whenever you can, maybe store the company currency as a hidden field in the form. 
So it becomes really easy to toggle `exchange_rate` display using display depends on `doc. account_currency != parent.company_currency` in the schema specification only.